File: safestep/utils/formatters.py
def format_currency(value: float) -> str:
    return f"£{value:,.0f}"


def format_percent(value: float) -> str:
    return f"{value:.1%}"


def format_number(value: float) -> str:
    return f"{value:,.0f}"


def format_ratio(value: float) -> str:
    return f"{value:.2f}x"
    

File: site/src/lib/waitwise/formatters.ts
export function formatCurrency(value: number): string {
  if (!Number.isFinite(value)) return "£0";
  return `£${value.toLocaleString("en-GB", {
    maximumFractionDigits: 0,
  })}`;
}

export function formatPercent(value: number): string {
  if (!Number.isFinite(value)) return "0.0%";
  return `${(value * 100).toFixed(1)}%`;
}

export function formatNumber(value: number): string {
  if (!Number.isFinite(value)) return "0";
  return value.toLocaleString("en-GB", {
    maximumFractionDigits: 0,
  });
}

export function formatRatio(value: number): string {
  if (!Number.isFinite(value)) return "0.00x";
  return `${value.toFixed(2)}x`;
}